rslib/src/search/mod.rs

```diff
@@ -363,7 +363,7 @@ fn card_order_from_sort_column(column: Column, timing: SchedTimingToday) -> Cow<
         )
         .into(),
         Column::Deck => "(select pos from sort_order where did = c.did) asc".into(),
-        Column::Due => "c.type asc, c.due asc".into(),
+        Column::Due => format!("c.type asc, (case when c.due > 1000000000 then due else (due - {}) * 86400 + {} end) asc", timing.days_elapsed, TimestampSecs::now().0).into(),
```

Comment:
The original motivation for sorting by c.type first was so that new cards and reviews don't end up getting mixed together if you have a card due on day 100 and a new card with #100. Because we're converting review dates to timestamps now, perhaps we can drop the initial type sorting here and below?

Reply:
Makes sense. One minor thing that might cause confusion is how new cards will get mixed with other types.
![image](https://github.com/ankitects/anki/assets/41397710/82e3420c-660f-4bb8-8d4f-7ec42612bd30)

Reply:
Hmm, I didn't expect that - I thought that since the due dates are being converted to timestamps, they'd all come after the new numbers. Any idea why that's happening?

Reply:
We're treating new card positions as days and converting them like we do for review cards. The cards with due date 2025-07-22 in my screenshot have the same due value (2683).

Reply:
Adding a check to avoid converting new card positions should work. Will push a fix.

Reply:
Ah, that makes sense - I feel silly for having overlooked that :-) Thanks Abdo.